src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,169 @@
+        htmlElement.scrollTop = 0
+    }
+    return iframe
+}
+const enableLivePreview = () => {
+    let hoveredElement: HTMLElement | null
+    let popupTimeout: ReturnType<typeof setTimeout> | null
+    let popper: Instance | null = null
+    const previewIframe = createPreviewIframe()
+    document.addEventListener('mouseover', (e: Event) => {
+        const target = e.target as HTMLElement
+        const isPageRef = target.classList.contains('rm-page-ref')
+        const isPageRefTag = target.classList.contains('rm-page-ref-tag')
+        // remove '#' for page tags
+        const text = isPageRefTag ? target.innerText.slice(1) : target.innerText
+        if (isPageRef) {
+            hoveredElement = target
+            const url = Navigation.getPageUrlByName(text)
+            const isAdded = (pageUrl: string) => !!document.querySelector(`[src="${pageUrl}"]`)
+            const isVisible = (pageUrl: string) =>
+                (document.querySelector(`[src="${pageUrl}"]`) as HTMLElement)?.style.opacity === '1'
+            if ((!isAdded(url) || !isVisible(url)) && previewIframe) {
+                previewIframe.src = url
+                previewIframe.style.height = '500px'
+                previewIframe.style.width = '500px'
+                previewIframe.style.pointerEvents = 'none'
+            }
+            if (!popupTimeout) {
+                popupTimeout = window.setTimeout(() => {
+                    if (previewIframe) {
+                        previewIframe.style.opacity = '1'
+                        previewIframe.style.pointerEvents = 'all'
+
+                        popper = createPopper(target, previewIframe, {
+                            placement: 'right',
+                            modifiers: [
+                                {
+                                    name: 'preventOverflow',
+                                    options: {
+                                        padding: {top: 48},
+                                    },
+                                },
+                                {
+                                    name: 'flip',
+                                    options: {
+                                        boundary: document.querySelector('#app'),
+                                    },
+                                },
+                            ],
+                        })
+                    }
+                }, 300)
+            }
+        }
+    })
+    document.addEventListener('mouseout', (e: MouseEvent) => {
+        const target = e.target as HTMLElement
+        const relatedTarget = e.relatedTarget as HTMLElement
+        const iframe = document.getElementById('roam-toolkit-preview-iframe') as HTMLIFrameElement
+        if (
+            (hoveredElement === target && relatedTarget !== iframe) ||
+            (target === iframe && relatedTarget !== hoveredElement) ||
+            !document.body.contains(hoveredElement)
```

Comment:
uh, this needs to be a function with the meaningful name. It's subexpressions need to be variables